packages/grid/_modules_/grid/components/DensityPicker.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import Button from '@material-ui/core/Button';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import { useIcons } from '../hooks/utils/useIcons';
+import { ApiContext } from './api-context';
+import { DensityTypes, Density } from '../models/gridOptions';
+import { useGridSelector } from '../hooks/features/core/useGridSelector';
+import { optionsSelector } from '../hooks/utils/useOptionsProp';
+import { useGridState } from '../hooks/features/core/useGridState';
+
+interface DensityConfig {
+  density: Density;
+  headerHeight: number;
+  rowHeight: number;
+}
+
+interface DensityOption {
+  icon: React.ReactElement;
+  label: DensityTypes;
+}
+
+const DENSITY_DIVISOR = 2;
+const DENSITY_MULTIPLIER = 1.5;
+
+export const DensityPicker = React.memo(function DensityPicker() {
+  const api = React.useContext(ApiContext);
+  const options = useGridSelector(api, optionsSelector);
+  const [, setGridState, forceUpdate] = useGridState(api!);
+  const initialDensityConfig = React.useRef<DensityConfig>({
+    density: options.density,
+    headerHeight: options.headerHeight,
+    rowHeight: options.rowHeight,
+  });
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const icons = useIcons();
+
+  const DensityShortIcon = icons!.densityShort!;
+  const DensityMediumIcon = icons!.densityMedium!;
+  const DensityTallIcon = icons!.densityTall!;
```

Comment:
Density is normally low/medium/high (size is small/medium/large)

Reply:
<img width="534" alt="Capture d’écran 2020-11-23 à 20 54 58" src="https://user-images.githubusercontent.com/3165635/100008777-2ad44780-2dce-11eb-9f92-ab4dab1a3b36.png">

https://material-density.glitch.me/

vs.

https://material-ui.com/components/tables/#dense-table `<Table size="small">`

Reply:
What if we call the prop `size`?

Reply:
"density" is more accurate, but if consistency with the Table is important, then sure. (Is it a prop though?)

Reply:
Looking at https://next.material-ui.com/customization/density/#main-content, we use the two names interchangeably. I would be in favor of using `size` everywhere. I think that it will be simpler.

Reply:
It's mostly either `margin: dense`, or `size: small`, with one instance of `dense: true`, and one of `variant: dense`. 

Of the first two, `size: small` is closest, even if it doesn't quite describe the purpose.

Reply:
Ok so I kept the prop name `density`, should I change it to `size` and also if yes then the same will apply to the button label right?

Reply:
I personally like the approach of using:

- "density" for the copywriting.
- `size?: "small" | "medium" | "large"` for the code API.

Reply:
The hook is called `useDensity`, the component is `densityPicker` and the label in the toolbar is `density` so why would we name it size in the prop? 
We should call it density as it's less confusing for us as well as for users and also, it's easier to find and understand in the list of props. Button density in the toolbar matches the density prop, it's logical and straightforward.

Otherwise, we can rename everything as `useSize`, `SizePicker`, and `size` in the label, and have a size property.
It doesn't make sense to me to have something named `useFoo`, `Foo` and expose it as `Bar`

Reply:
> The hook is called useDensity, the component is densityPicker and the label in the toolbar is density so why would we name it size in the prop?

We can rename them all to "size".

I think that size is superior in the code because more generic and allow consistency. Why don't we call `<Button size="small">` and not `<Button density="dense">`?  https://ant.design/components/table/#components-table-demo-size, https://material-ui.com/components/tables/#dense-table.